src/unix/linux-core.c

```diff
@@ -587,7 +600,10 @@ static int read_times(FILE* statfile_fp,
     ts.irq  = irq * multiplier;
     ci[num++].cpu_times = ts;
   }
-  assert(num == numcpus);
+
+  if (num != numcpus) {
+    return UV_EAGAIN;
+  }
```

Comment:
Style:
```suggestion
  if (num != numcpus)
    return UV_EAGAIN;
```
Substance: I'd say it's better to return `num` and let the caller decide what to do. Applies to `read_models()` and `read_speeds()` too, although I think the latter is infallible (never errors.)

Reply:
`read_models` and `read_speeds` both fill the cpu_info slots that are unable to read therefore they are infallible on online/offline race. I updated `read_times` to return `numread`.